src/boot/efi/ticks.c

```diff
@@ -17,7 +18,35 @@ static uint64_t ticks_read_arch(void) {
 }
 
 static uint64_t ticks_freq_arch(void) {
-        return 0;
+        /* Detect TSC frequency from CPUID information if available. */
+
+        unsigned max_leaf, ebx, ecx, edx;
```

Comment:
uint32_t?

Reply:
cpuid takes `unsigned int`, not `uint32_t`.

Reply:
still weird though, since eax is definitely 32bit wide...